common/remote/sockfile.cpp

```diff
@@ -4280,6 +4330,64 @@ class CRemoteFileServer : public CInterface, implements IRemoteFileServer, imple
         return false;
     }
 
+    bool cmdSetThrottle(MemoryBuffer & msg, MemoryBuffer & reply,CRemoteClientHandler &client)
+    {
+        CriticalBlock b(setThrottleCrit);
+        try
+        {
+            unsigned _throttleLimit, _throttleDelayMs, _throttleCPULimit;
+            msg.read(_throttleLimit);
+            msg.read(_throttleDelayMs);
+            msg.read(_throttleCPULimit);
+            if (_throttleLimit > throttleLimit)
+            {
+                PROGLOG("Increasing throttleLimit from %d to %d", throttleLimit, _throttleLimit);
+                unsigned diff = _throttleLimit - throttleLimit;
+                throttlesem.signal(diff);
+                throttleLimit = _throttleLimit;
+                // NB: If throttling was off, this doesn't effect transactions in progress, i.e. will only throttle new transactions coming in.
+            }
+            else
+            {
+                unsigned diff = throttleLimit - _throttleLimit;
+                unsigned prevLimit = throttleLimit;
+                PROGLOG("Reducing throttleLimit from %d to %d", throttleLimit, _throttleLimit);
+                if (0 == _throttleLimit)
+                {   /* disabling - set limit immediately to let all new transaction through
+                     * but still need to consume the signals, to avoid issues if throttling changed again
+                     */
+                    throttleLimit = 0;
+                }
+
+                // NB: This is not expected to take long
+                CCycleTimer timer;
+                while (diff)
+                {
+                    if (throttlesem.wait(1000))
```

Comment:
I'm not sure about this line.  If the throttle limit is set to 0, and there are several items waiting, could this grab all the signals, and leave those items forever waiting?